synapse/cores/common.py

```diff
@@ -1573,7 +1577,7 @@ def addSpliceFd(self, fd):
             core.addSpliceFd(fd)
         '''
         def save(mesg):
-            fd.write(s_msgpack.en(mesg))
+            fd.write(s_msgpack.en(mesg[1]['mesg']))
```

Comment:
use ``.get()`` instead of ``[]``

Reply:
I did it this way so that save would either write the encapsulated message or fail. If I use .get, I will have to check that it actually contains a message to avoid writing `b'\xc0'` to the fd.

Reply:
this guys content is gonna end up getting passed to splices() (most likely) which will catch an exception raised by a bad message.

Reply:
Yep - but is it a problem if msgpack'd `None` is written to the fd vs just raising exception and continuing?

Reply:
We decided to keep this as-is.

Reply:
we talked though this and should keep the existing behavior and ignore my comment here